File: OS/Denominaciones.c
/*
Programa:		Denominaciones.c

Objetivo: 			El programa debe imprimir cuántos billetes y monedas en efectivo debe entregar
				un cajero para al dueño del cheque. El cajero tan solo dispone de billetes de 50 y
				20 pesos, así como monedas de 10, 5, 2 y 1 peso.

Autor:			Michael Duarte
Fecha:			Septiembre 2012
*/

#include <stdio.h>

int denom[]={50,20,10,5,2,1};
void diffDenom(int,int[], int[],int);
void imprimirDenom(int[],int[],int);


int main(){
	int cantidad,  		size=sizeof(denom)/sizeof(int);
	int exit[size],		i;
	
	//inicializamos arreglo de salida con 0s
	for(i=0;i<size;i++)
		exit[i]=0;
	
	
	printf("Introduce cantidad del cheque: ");
	scanf("%d",&cantidad);
	
	//buscaremos variedad de denominaciones necesitaremos
	printf("Denominaciones para: %d\n",cantidad);
	diffDenom(cantidad,denom,exit,size);
	
	//imprimiremos la variedad de denominaciones
	imprimirDenom(denom,exit,size);
	
	return 0;
}

void diffDenom(int cantidad, int in[],int exit[], int size){
	int temp;
	int i = 0;
	for(i=0;i<size&&cantidad>0;i++){
		if((temp=cantidad/in[i])>0){
			//printf("%d x $%d\n",temp,in[i]);
			exit[i]=temp;
			cantidad-=temp*in[i];
		}
	}
}

void imprimirDenom(int denom[], int unidades[], int size){
	int i;
	for(i=0;i<size;i++){
		if(unidades[i]>0){
			printf("\t%d x $%d\n",unidades[i],denom[i]);
		}
	}
	
}
	

File: OS/Denominaciones.java
/*
Programa:		Denominaciones.java
Objetivo:		Dado una cantidad de dinero, se debera imprimir el numero de 
			monedas/billetes de cada denominación existente.s

Autor:			Michael Duarte
Fecha:			Septiembre 2012

*/

public class Denominaciones{

	public static void main(String[] arg){
	Denominaciones b=new Denominaciones();
	int cantidad=183;
	if(arg.length==1){
		try{
		      cantidad=Integer.parseInt(arg[0]);
		}catch(Exception e){
			System.out.printf("Parametros deben ser numericos\n"); 
			System.exit(0);	
		}
	}
	System.out.printf("Cantidad: %d\n",cantidad);
	int[] denom=new int[]{50,20,10,5,2,1};
	int[] salida=b.diffDenom(cantidad,denom);
	b.salida(denom,salida);
	
	}
	
	public int[] diffDenom(int cantidad, int[] in){
		int temp;
		int[] exit=new int[in.length];
		for(int i=0;i<in.length&&cantidad>0;i++){
			if((temp=cantidad/in[i])>0){
				exit[i]=temp;
				cantidad-=temp*in[i];
			}
		}
		return exit;
		
	}
	
	public void salida(int[] denom,int[] res){
		for(int i=0;i<denom.length;i++){
			if(res[i]>0){
				System.out.printf("\t%d x $%d\n",res[i],denom[i]);
			}
		}
	}

}